File: python/appkey_demo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys,os
import base64
import time
import json

from urlparse import urlparse
from com.aliyun.api.gateway.sdk import client
from com.aliyun.api.gateway.sdk.http import request
from com.aliyun.api.gateway.sdk.common import constant
import traceback
import urllib2
import base64

def get_img_base64(img_file):
    with open(img_file, 'rb') as infile:
        s = infile.read()
        return base64.b64encode(s) 


def predict(url, app_key, app_secret, img_base64, kv_config, old_format):
        statTime = time.time()
        cli = client.DefaultClient(app_key=app_key, app_secret=app_secret)
        if not old_format:
            param = {}
            param['image'] = img_base64
            if kv_config is not None:
                param['configure'] = json.dumps(kv_config)
            body = json.dumps(param)
        else:
            param = {}
            pic = {}
            pic['dataType'] = 50
            pic['dataValue'] = img_base64
            param['image'] = pic
    
            if kv_config is not None:
                conf = {}
                conf['dataType'] = 50
                conf['dataValue'] = json.dumps(kv_config) 
                param['configure'] = conf

    
            inputs = { "inputs" : [param]}
            body = json.dumps(inputs)

        url_ele = urlparse(url)
        host = 'http://' + url_ele.hostname
        path = url_ele.path 
        headers = {}
        req_post = request.Request(host=host, protocol=constant.HTTP, url=path, headers = headers, method="POST", time_out=6000)
        req_post.set_body(bytearray(source=body, encoding="utf8"))
        req_post.set_content_type(constant.CONTENT_TYPE_STREAM)
        stat,header, content = cli.execute(req_post)
        endTime = time.time()

        return stat, dict(header) if header is not None else {}, content


def demo():
    app_key = '你的APP_KEY'
    app_secret = '你的APP_SECRET'
    url = 'http://dm-51.data.aliyun.com/rest/160601/ocr/ocr_idcard.json'
    img_file = '图片文件路径'
    #如果输入带有inputs, 设置为True，否则设为False
    is_old_format = True  
    config = {'side':'face'}
    #如果没有configure字段，config设为None
    #config = None

    img_base64data = get_img_base64(img_file)
    stat, header, content = predict( url, app_key, app_secret, img_base64data, config, is_old_format)
    if stat != 200:
        print 'Http status code: ', stat
        print 'Error msg in header: ', header['x-ca-error-message'] if 'x-ca-error-message' in header else ''
        print 'Error msg in body: ', content
        exit()
    if is_old_format:
        result_str = json.loads(content)['outputs'][0]['outputValue']['dataValue']
    else:
        result_str = content

    print result_str;
    #result = json.loads(result_str)

if __name__ == '__main__':
    demo()

File: php/appkey_demo.php
<?php
    include_once 'api-gateway-demo-sign-php/Util/Autoloader.php';
    
    
    function doPost($url, $appKey, $appSecret, $bodyContent) {
        //域名后、query前的部分
        $urlEles = parse_url($url);
        $host = $urlEles["scheme"] . "://" . $urlEles["host"];
        $path = $urlEles["path"];
        $request = new HttpRequest($host, $path, HttpMethod::POST, $appKey, $appSecret);
    
        //设定Content-Type，根据服务器端接受的值来设置
        $request->setHeader(HttpHeader::HTTP_HEADER_CONTENT_TYPE, ContentType::CONTENT_TYPE_JSON);
    
        //设定Accept，根据服务器端接受的值来设置
        $request->setHeader(HttpHeader::HTTP_HEADER_ACCEPT, ContentType::CONTENT_TYPE_JSON);
    
        //注意：业务body部分，不能设置key值，只能有value
        if (0 < strlen($bodyContent)) {
            $request->setHeader(HttpHeader::HTTP_HEADER_CONTENT_MD5, base64_encode(md5($bodyContent, true)));
            $request->setBodyString($bodyContent);
        }
    
        //指定参与签名的header
        $request->setSignHeader(SystemHeader::X_CA_TIMESTAMP);
        $response = HttpClient::execute($request);
        return $response;
    }
    
    $appKey = "你的APPKEY";
    $appSecret = "你的APPSECRET";
    $url = "http://dm-51.data.aliyun.com/rest/160601/ocr/ocr_idcard.json";
    $file = "图片路径/图片URL"; 
    
    //如果没有configure字段，configure设为空
    $configure = array(
        "side" => "face"
    );
    //$configure = array()

    if(substr($file, 0, 4) == "http") {
        $base64 = $file; 
    }else if($fp = fopen($file, "rb", 0)) { 
        $binary = fread($fp, filesize($file)); // 文件读取
        fclose($fp); 
        $base64 = base64_encode($binary); // 转码
    }

    $request = array(
        "image" => "$base64"
    );
    if(count($configure) > 0){
        $request["configure"] = json_encode($configure);
    }
    $body = json_encode($request);
    $response = doPost($url, $appKey, $appSecret, $body);
    $stat = $response->getHttpStatusCode();
    if($stat == 200){
        $result_str = $response->getBody();
        printf("result is :\n %s\n", $result_str);

    }else{
        printf("Http error code: %d\n", $stat);
        printf("Error msg in body: %s\n", $response->getBody());
        printf("header: %s\n", $response->getHeader());
}
    //var_dump($response);
